src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -484,5 +484,7 @@ private void InitializeCachedOptions(OptionSet solutionOptions, CancellationToke
 
             Interlocked.CompareExchange(ref _cachedOptions, newAsyncLazy, comparand: null);
         }
+
+        internal Task<ImmutableDictionary<string, string>> GetAnalyzerOptionsAsync(CancellationToken cancellationToken) => DocumentState.GetAnalyzerOptionsAsync(cancellationToken);
```

Comment:
consider wrapping.

Reply:
wouldn't it make sense for this to be public?

Reply:
Yes, I expect it will be public at some point. Still unclear to me if that's the right return type, so we'll leave it internal for _now_.